PhysicsTools/SelectorUtils/python/VIDSelectorBase.py

```diff
@@ -13,6 +13,34 @@
 ROOT.gSystem.Load("libDataFormatsFWLite.so");
 ROOT.FWLiteEnabler.enable()
 
+# define some C++ helpers that are only used in this VID selector class and deriving classes
+ROOT.gInterpreter.Declare("""
+#include "FWCore/ParameterSetReader/interface/ParameterSetReader.h"
+
+template <class PhysObj>
+struct MakeVersionedSelector {
+  MakeVersionedSelector() {}
+
+  VersionedSelector<edm::Ptr<PhysObj> > operator()(const std::string& pset, const std::string& which_config) {
+    const edm::ParameterSet& temp = edm::readPSetsFrom(pset)->getParameter<edm::ParameterSet>(which_config);
+    return VersionedSelector<edm::Ptr<PhysObj> >(temp);
+  }
+
+  VersionedSelector<edm::Ptr<PhysObj> > operator()() { return VersionedSelector<edm::Ptr<PhysObj> >(); }
+};
+
+template <class Collection,
+          class InPhysObj = typename Collection::value_type,
+          class OutPhysObj = typename Collection::value_type>
+struct MakePtrFromCollection {
+  edm::Ptr<OutPhysObj> operator()(const Collection& coll, unsigned idx) {
+    edm::Ptr<InPhysObj> temp(&coll, idx);
+    return edm::Ptr<OutPhysObj>(temp);
+  }
+};
+""")
```

Comment:
Does the dictionary generation get tested in unit tests?

Reply:
Unit tests can be added to python files along
```py
if __name__ == "__main__:
    VIDMuonSelector()
```
For a concrete example see e.g. framework python files
https://github.com/cms-sw/cmssw/blob/30bb8dca17f6bc7b7c6b77283f92b9cae3c745b8/FWCore/ParameterSet/python/Config.py#L1762-L1764

In this case I'd add one for each of the `VID(Muon|Electron|Photon)Selector.py` file.